pandas/core/internals/managers.py

```diff
@@ -1679,6 +1681,41 @@ def create_block_manager_from_arrays(
         raise construction_error(len(arrays), arrays[0].shape, axes, e)
 
 
+def create_block_manager_from_array(
```

Comment:
can you not *just* call create_block_manager_from_arrays? and if not why not?

Reply:
Because `create_block_manager_from_arrays` (and `_form_blocks`, and `_stack_arrays`) is written to receive a list of 1-dimensional columns, and create copy of them. My latest commit adjusts these functions to avoid copying when necessary.